### ISSUE TITLE
Showing only 7 Days in a week

### ISSUE THREAD

**Issue by usmankhan1990** [kizitonwose/CalendarView#253]
I want to display just 7 Days of week, means any specific date (starts from Sunday) will show only 7 Days. So there will be no Scrolling also as I only need 7 Days to show. 

This is how I am doing a setup:

        ```binding.exSevenCalendar.apply {
            val dayWidth = dm.widthPixels / 7
            val dayHeight = (dayWidth * 2)
            daySize = Size(dayWidth, dayHeight)
        }

        val currentMonth = YearMonth.now()
        binding.exSevenCalendar.setup(
            currentMonth,
            currentMonth.plusMonths(0),
            DayOfWeek.SUNDAY
        )
        binding.exSevenCalendar.scrollToDate(LocalDate.of(2021, 1, 29))```

So like 29th Jan 2021 date will show only 7 Numbers and I don't need any other Day values. For example out of 31 Days, I only need 29th Jan Week Days.

**Comment by usmankhan1990** [comment]
Why I want to do it, because I need to disable Scrolling, I am using one of your code from old scrolling.
```
binding.exSevenCalendar.addOnItemTouchListener(object : RecyclerView.SimpleOnItemTouchListener() {
            override fun onInterceptTouchEvent(rv: RecyclerView, e: MotionEvent): Boolean {
                return rv.scrollState == RecyclerView.SCROLL_STATE_DRAGGING
            }
        })
```

If I am using above touch Listener, then it `onBind` will not call. I am following your Example 7 code. That is why I thought instead of disabling scroll, I need only 7 Days to display, So if I find any way to get only 7 Days out of full month.

**Comment by Kim-DongGeun** [comment]
follow this step!
![image](https://user-images.githubusercontent.com/12513110/103472451-1c1b8000-4dd1-11eb-8c69-cd350f52540d.png)

**Comment by usmankhan1990** [comment]
@Kim-DongGeun Thank you for your response. I tried above but still getting all other Months days. And also my week is also not starting from SUNDAY. Please check my below code:

```
override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding = Example7FragmentBinding.bind(view)

        val dm = DisplayMetrics()
        val wm = requireContext().getSystemService(Context.WINDOW_SERVICE) as WindowManager
        wm.defaultDisplay.getMetrics(dm)
        binding.exSevenCalendar.apply {
            val dayWidth = dm.widthPixels / 7
            val dayHeight = (dayWidth * 2)
            daySize = Size(dayWidth, dayHeight)
        }

        class DayViewContainer(view: View) : ViewContainer(view) {
            val bind = AppointmentCalendarDayBinding.bind(view)
            lateinit var day: CalendarDay

            fun bind(day: CalendarDay) {
                this.day = day
                bind.exSevenDateText.text = dateFormatter.format(day.date)
                bind.exSevenDayText.text = dayFormatter.format(day.date)
                bind.dotImg.visibility = View.VISIBLE
                bind.exSevenDateText.setTextColor(view.context.getColorCompat(if (day.date == selectedDate) R.color.example_4_grey else R.color.example_4_grey))
            }
        }

        binding.exSevenCalendar.dayBinder = object : DayBinder<DayViewContainer> {
            override fun create(view: View) = DayViewContainer(view)
            override fun bind(container: DayViewContainer, day: CalendarDay) = container.bind(day)
        }


        binding.exSevenCalendar.updateMonthConfiguration(

            inDateStyle = InDateStyle.ALL_MONTHS,
            maxRowCount = 1,
            hasBoundaries = false
        )

        val currentMonth = YearMonth.now()
        binding.exSevenCalendar.setup(
            currentMonth,
            currentMonth.plusMonths(1),
            DayOfWeek.SUNDAY
        )

        binding.exSevenCalendar.scrollToDate(LocalDate.of(2021, 1, 23))
//        binding.exSevenCalendar.suppressLayout(true)
    }
```
By doing scrolling, other month Days are also coming. My requirement is to only show 7 DAYS of any selected Date, starting from SUNDAY as the First Day of the week. Kindly guid me for this.

<img width="478" alt="Screen Shot 2021-01-03 at 10 14 07 AM" src="https://user-images.githubusercontent.com/8194315/103472915-6c341b80-4dac-11eb-95f7-c92b07b0ba70.png">

**Comment by usmankhan1990** [comment]
I used the below code also, but didn't work:

        ```
binding.exSevenCalendar.updateMonthConfiguration(

            inDateStyle = InDateStyle.FIRST_MONTH,
            maxRowCount = 1,
            hasBoundaries = true
        )

```

I am getting wrong FIRST_DAY_OF_WEEK format as Saturday (Which I need SUNDAY) and also I can see other days in scrolling. Why I need only 7 Days, just to avoid scrolling. I need to Disable scrolling so I thought if any how I can show only 7 Days, then no scrolling wouldn't be available.

**Comment by Kim-DongGeun** [comment]
@usmankhan1990 if you want to start SUNDAY, you should replace this code 
![image](https://user-images.githubusercontent.com/12513110/103475017-d371c000-4dec-11eb-9bbb-aa34cd904afe.png)
in Extension.kt
if firstDayOfWeek is SUNDAY, start of your calendar will be Sunday.

and i don't know how to display only 7 days
sorry bro
but i suppose that you can show only 1 month using this code
![image](https://user-images.githubusercontent.com/12513110/103475244-d7064680-4dee-11eb-9f4b-242c8ee44d19.png)
Set startMonth and endMonth to currentMonth

**Comment by asthagarg2428** [comment]
@usmankhan1990 Were you able to achieve 'showing only 7 days a week?' Likewise I'm setting my daySize to little less and have margin in calendarView from left and right, but in the remaining space, the next month's dates are coming.
Any help?
I want to have left and right spacing to CalendarView still showing 7 days of week completely within that

**Comment by kizitonwose** [comment]
@asthagarg2428 Did you subtract the margins before dividing the width of the calendar to get the width for each cell?

**Comment by asthagarg2428** [comment]
> @asthagarg2428 Did you subtract the margins before dividing the width of the calendar to get the width for each cell?

Now I'm using monthPadding.. it's working now

**Comment by asthagarg2428** [comment]
Now I'm stuck at another case where in I want to show maximum of 5 rows in each month and if any date is remaining, so it should go with next month.
So o achieve this, I used         
`calendarView.updateMonthConfiguration(InDateStyle.ALL_MONTHS, OutDateStyle.END_OF_ROW,5,true);`

But the problem now coming is that if 1st of March is shown with Feb month, now when I swipe right to March Calendar, 1st March is not coming.
So basically, my month should start with 1st Date.
Hope I'm able to convey my requirement..
